### PR TITLE
feat(docs): configure Storybook Docs

### PR DESCRIPTION
**Pull request by connor-baer**
Addresses #463.

## Purpose

This PR lays the foundations to migrate our documentation to [Storybook Docs](https://github.com/storybookjs/storybook/issues/7430).

## Approach and changes

- `addon-options` has been [deprecated since Storybook 5](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#options-addon-deprecated) and has been removed.
- Storybook added a new `component` parameter for stories to automagically enable some features such as a props table. This change was ✨ [codemodded](https://github.com/storybookjs/storybook/blob/next/lib/codemod/README.md#add-component-parameters).
- Storybook introduced a new [component story format](https://medium.com/storybookjs/component-story-format-66f4c32366df). This change was ✨ [codemodded](https://github.com/storybookjs/storybook/blob/next/lib/codemod/README.md#storiesof-to-csf).
- The above codemods required the Storybook group enums to be removed.
- [Storybook DocsPage](https://medium.com/storybookjs/storybook-docspage-e185bc3622bf) is what this PR is all about. The required configurations were added and I took the opportunity to simplify the Storybook configs.
- The [Storybook theme](https://storybook.js.org/docs/configurations/theming/) was updated with the Circuit UI branding.

Captura de tela 2019-10-20 01 08 51

_The migration of the docs and removal of Docz will happen in a separate PR._

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Comment by connor-baer** [comment]
@voronianski + @lucent1090, could you review this PR, please? Once it's merged, we can move forward with Storybook Docs and the dependency upgrades for v2 :)

File: package.json [hunk 1]
```diff
@@ -21,12 +21,11 @@
     "test:unit:output": "jest --json --outputFile=jest-test-results.json",
     "create-component": "foundry run plop component",
     "static-styles": "cross-env BABEL_ENV=static babel-node ./scripts/static-styles/cli.js",
-    "prestart": "yarn test:unit",
-    "start": "start-storybook -p 6006",
-    "prebuild": "yarn test:unit:output",
+    "prestart": "yarn test:unit:output",
```
Comment (voronianski) [inline package.json]: thank you for this!

File: src/components/LoadingButton/LoadingButton.story.js [hunk 2]
```diff
@@ -14,55 +14,63 @@
...
+export default {
+  title: 'Components|Button/LoadingButton',
+
+  parameters: {
+    component: LoadingButton,
+    jest: ['LoadingButton']
+  }
+};
+
+export const loadingButtonWithSuccessAnimation = () => (
```
Comment (voronianski) [inline src/components/LoadingButton/LoadingButton.story.js]: is it fine that most of the exported components are starting from lowercase letter?

Comment (voronianski) [reply src/components/LoadingButton/LoadingButton.story.js]: ok, it's according to their docs, nvmnd

**Pull request merged**